src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEModuleSymbol.cs

```diff
@@ -641,47 +641,51 @@ internal NamedTypeSymbol LookupTopLevelMetadataType(ref MetadataTypeName emitted
         }
 
         /// <summary>
-        /// If this module forwards the given type to another assembly, return that assembly;
-        /// otherwise, return null.
+        /// Returns a tuple of the assemblies this module forwards the given type to.
         /// </summary>
         /// <param name="fullName">Type to look up.</param>
-        /// <returns>Assembly symbol or null.</returns>
+        /// <returns>A tuple of the forwarded to assemblies.</returns>
         /// <remarks>
-        /// The returned assembly may also forward the type.
+        /// The returned assemblies may also forward the type.
         /// </remarks>
-        internal AssemblySymbol GetAssemblyForForwardedType(ref MetadataTypeName fullName)
+        internal (AssemblySymbol FirstSymbol, AssemblySymbol SecondSymbol) GetAssembliesForForwardedType(ref MetadataTypeName fullName)
         {
             string matchedName;
-            AssemblyReferenceHandle assemblyRef = Module.GetAssemblyForForwardedType(fullName.FullName, ignoreCase: false, matchedName: out matchedName);
-            return assemblyRef.IsNil ? null : this.GetReferencedAssemblySymbol(assemblyRef);
+            (int firstIndex, int secondIndex) = this.Module.GetAssemblyRefsForForwardedType(fullName.FullName, ignoreCase: false, matchedName: out matchedName);
+
+            var firstSymbol = firstIndex < 0 ? null : GetReferencedAssemblySymbol(firstIndex);
+            var secondSymbol = secondIndex < 0 ? null : GetReferencedAssemblySymbol(secondIndex);
+
+            return (firstSymbol, secondSymbol);
         }
 
         internal IEnumerable<NamedTypeSymbol> GetForwardedTypes()
         {
-            foreach (KeyValuePair<string, AssemblyReferenceHandle> forwarder in Module.GetForwardedTypes())
+            foreach (KeyValuePair<string, (int FirstIndex, int SecondIndex)> forwarder in Module.GetForwardedTypes())
             {
-                var assemblySymbol = this.GetReferencedAssemblySymbol(forwarder.Value);
-                if ((object)assemblySymbol == null)
+                if (forwarder.Value.FirstIndex < 0)
```

Comment:
>if (forwarder.Value.FirstIndex < 0) [](start = 16, length = 35)

I do not believe it is possible for this condition to ever be true. We should assert that and remove the ```if```.